### PR TITLE
fix: web socket reconnect throttle

### PR DESCRIPTION
**Pull request by robot9706**
When the WebSocket disconnects (due to an invalid or expired token) it tries to reconnect multiple times per second.
Implemented throttling to prevent this.

### CODE REVIEW

File: web/crux/src/websockets/dyo.ws.adapter.ts [hunk 1]
```diff
@@ -124,25 +149,6 @@ export default class DyoWsAdapter extends AbstractWsAdapter {
     return super.close(server)
   }
 
-  private async authorize(client: WsClient): Promise<boolean> {
-    const ctx = new ExecutionContextHost([client.connectionRequest], null, this.authorize)
-    let authorized = false
-    try {
-      authorized = await this.authGuard.canActivate(ctx)
-    } catch {
-      /* empty */
```
Comment (polaroi8d) [inline web/crux/src/websockets/dyo.ws.adapter.ts:133]: Thanks for deleted the empty catch.

File: web/crux-ui/src/websockets/websocket-client.ts [hunk 2]
```diff
@@ -234,71 +234,74 @@ class WebSocketClient {
   private createConnectionAttempt(): Promise<boolean> {
     this.lastAttempt = Date.now()
 
-    return new Promise<boolean>(resolve => {
-      this.logger.debug('Connecting...')
-      let resolved = false
+    const attempt = () =>
+      new Promise<boolean>(resolve => {
+        this.logger.debug('Connecting...')
```
Comment (polaroi8d) [inline web/crux-ui/src/websockets/websocket-client.ts:239]: Who? to who? Can we add a little bit more verbose log message here?

File: web/crux-ui/src/websockets/websocket-client.ts [hunk 3]
```diff
@@ -234,71 +234,74 @@ class WebSocketClient {
...
+          if (!resolved) {
+            resolved = true
+            resolve(true)
+          }
 
-      const onOpen = () => {
-        if (!resolved) {
-          resolved = true
-          resolve(true)
+          this.logger.info('Connected')
```
Comment (polaroi8d) [inline web/crux-ui/src/websockets/websocket-client.ts:248]: Can we improve the same here?

**Review by polaroi8d — APPROVED** [review]
You could add (web) scope to the pull request. Minor reviews, otherwise okay for me.

**Review by m8vago — CHANGES_REQUESTED** [review]

**Review by m8vago — APPROVED** [review]

**Pull request merged**